### PR TITLE
Encryption impl with ElGamal

### LINKED ISSUE #19: Add encryption functionalities for value & blinder fields and refactor the code for them
As discussed with @toghrulmaharramov we need to add the encryption step for the value and the blinder fields for the `Bid` generation.

The code actually doesn't have any logic implemented for this.

### PR DESCRIPTION
**Pull request by CPerezz**
We've edited the circuits and the implementation in
order to adapt the repo to ElGamal encryption schema.

Both the value and the blinder fields are now provided to
the gadgets but ASAP once the ElGmal schema is implemented
they will be decrypted and used instead of being provided.

Apart from that, the versions of the dependencies have been
updated and restrictions to the Bid generation have been added
in order to not surpass the maximum bid value allowed.

Closes #19

### CODE REVIEW

File: src/bid/bid.rs [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 use crate::score_gen::{compute_score, Score};
 use dusk_plonk::jubjub::{
     AffinePoint, ExtendedPoint, GENERATOR, GENERATOR_NUMS,
 };
 use dusk_plonk::prelude::*;
 use failure::Error;
-use poseidon252::sponge::sponge::sponge_hash;
-pub(crate) mod encoding;
-pub use encoding::StorageBid;
+use poseidon252::{cipher::PoseidonCipher, sponge::sponge::sponge_hash};
```
Comment (vlopes11) [inline src/bid/bid.rs:11]: `PoseidonCipher` is not used anywhere in this PR. Why is it imported?

File: src/bid/bid.rs [hunk 2]
```diff
@@ -24,21 +24,13 @@ pub struct Bid {
...
-    // b (blinder)
-    pub(crate) blinder: JubJubScalar,
-    // b_enc (encrypted blinder) // XXX: Scalar for now. Double check
-    pub(crate) encrypted_blinder: JubJubScalar,
-    // v (Bid value)
-    pub(crate) value: JubJubScalar,
-    // v_enc (encrypted_value)
-    pub(crate) encrypted_value: JubJubScalar,
+    // The encrypted value & blinder fields were previously encrypted with `ElGamal` schema.
+    pub(crate) encrypted_blinder: (AffinePoint, AffinePoint),
```
Comment (vlopes11) [inline src/bid/bid.rs:32]: Its unclear to me why the encrypted blinder is represented with 2 points. If its a poseidon cipher, should be either 3 scalars `Fq` or `PoseidonCipher` itself

File: src/bid/bid.rs (hunk 1 above)
Comment (CPerezz) [reply src/bid/bid.rs:11]: Yep, I should remove it, my bad. This was from the previous impl.

File: src/bid/bid.rs (hunk 2 above)
Comment (CPerezz) [reply src/bid/bid.rs:32]: As mentioned @toghrulmaharramov yesterday the encrypted values with ElGamal will be represented by to AffinePoints.

Comment (vlopes11) [reply src/bid/bid.rs:29]: Ok, but later on we may replace this for the encrypted structure to benefit from the homomorphic properties of elgamal cipher

**Pull request merged**